presto-main/src/main/java/io/prestosql/sql/gen/SqlTypeBytecodeExpression.java

```diff
@@ -40,14 +40,32 @@ public static SqlTypeBytecodeExpression constantType(CallSiteBinder callSiteBind
         return new SqlTypeBytecodeExpression(type, binding, BOOTSTRAP_METHOD);
     }
 
+    private static String generateName(Type type)
+    {
+        Class<?> fromJavaElementType = type.getJavaType();
```

Comment:
Instead of using the stack type, I would use the existing logic up to some size limit (maybe 20 bytes), and after that, use `type.getTypeSignature().getBase()` which will be ROW for this case.

Reply:
Sounds good. Updated the commit.